pyanaconda/dbus/observer.py

```diff
@@ -0,0 +1,229 @@
+    result = proxy.ListConnections()
+    print(result)
+
+    """
+
+    def __init__(self, service_name):
+        """Creates the DBus service observer.
+
+        :param service_name: a DBus name of a service
+        """
+        super().__init__(service_name)
+        self._proxies = dict()
+
+    def get_proxy(self, object_path):
+        """"Returns a proxy of the remote object."""
+        if not self._is_service_available:
+            raise DBusObserverError("Service %s is not available.",
+                                    self._service_name)
+
+        if object_path in self._proxies:
+            return self._proxies.get(object_path)
+
+        proxy = DBus.get_proxy(self._service_name, object_path)
+        self._proxies[object_path] = proxy
+        return proxy
+
+    def _service_name_appeared_callback(self, *args):
+        """Callback far the watch method."""
+        self._proxies = dict()
+        super()._service_name_appeared_callback(*args)
+
+    def _service_name_vanished_callback(self, *args):
+        """Callback for the watch method."""
+        self._proxies = dict()
+        super()._service_name_vanished_callback(*args)
+
+
+class DBusObjectObserver(DBusObserver):
+    """Observer of a DBus object.
+
+    This class is recommended to use when you are interested in only one
+    object provided by a service. You can specify the object path in the
+    __init__ method and access the proxy of the object in the proxy property.
+    The proxy is cached.
+
+    Usage:
+
+    observer = DBusObjectObserver("org.freedesktop.NetworkManager",
+                                  "org/freedesktop/NetworkManager/Settings")
+    observer.watch()
+    result = observer.proxy.ListConnections()
+    print(result)
+
+    """
+
+    def __init__(self, service_name, object_path):
+        """Creates an DBus object observer.
+
+        :param service_name: a DBus name of a service
+        :param object_path: a DBus path of an object
+        """
+        super().__init__(service_name)
+        self._proxy = None
+        self._object_path = object_path
+
+    @property
+    def proxy(self):
+        """"Returns a proxy of the remote object."""
+        if not self._is_service_available:
+            raise DBusObserverError("Service %s is not available.",
+                                    self._service_name)
+
+        if not self._proxy:
+            self._proxy = DBus.get_proxy(self._service_name,
+                                         self._object_path)
+
+        return self._proxy
+
+    def _service_name_appeared_callback(self, *args):
+        """Callback far the watch method."""
+        self._proxy = None
+        super()._service_name_appeared_callback(*args)
+
+    def _service_name_vanished_callback(self, *args):
+        """Callback for the watch method."""
+        self._proxy = None
+        super()._service_name_vanished_callback(*args)
+
+    def __str__(self):
+        return self._object_path
```

Comment:
The logs are awful when you try to do that. Usually you don't have to know the service name, because the object path is unique. You can always print `observer.service_name` when you need to do that.

Reply:
Maybe we could return DBusObjectObserver(service name, object path) for __repr__.

Reply:
Fixed with `__repr__`.